src/parsers/installrdf.js

```diff
@@ -105,6 +109,40 @@ export default class InstallRdfParser {
       this._getTopLevelNodeByTag('em:bootstrap')) === 'true';
   }
 
+  _getApplication() {
+    var targetApplications = this._getTopLevelNodesByTag(
+      'em:targetApplication');
+    var applications = [];
+
+    for (let targetApplication of targetApplications) {
+      let application = {};
+      let descriptionNodes = Array.prototype.slice.call(
+          targetApplication.childNodes)
+          .filter((node) => node.nodeName === 'Description');
+
+      if (descriptionNodes.length !== 1) {
+        throw new Error(singleLineString`<em:targetApplication> node should only
+          have a single descendant <Description>`);
+      }
+
+      for (var i in descriptionNodes[0].childNodes) {
+        let childNode = descriptionNodes[0].childNodes[i];
+        if (TARGET_APPLICATION_FIELDS.indexOf(childNode.localName) !== -1) {
+          application[childNode.localName] = this._getNodeValue(childNode);
+        }
+      }
+
+      if (application.id && application.maxVersion && application.minVersion) {
+        applications.push(application);
+      } else {
+        throw new Error(singleLineString`targetApplication must contain an id,
```

Comment:
Shouldn't this be an error message pushed onto the collector?

Also I think doing something a little more granular will help developers e.g. flag each field that's invalid separately so that it's clearer when just one thing is broken.